<?xml version="1.0"?>
<launch>
    <!-- declare arg to be passed in -->
    <arg name="team_side"     default="home" />
    <arg name="my_number"     default="1" />
    <arg name="ally_number"   default="2" />
    <arg name="orienter_path" default="/teamrocket_home/orienter" />

    <!-- using which side of the field we are on, determine the appropriate -->
    <!-- connections to the output of the orienter system -->
    <arg name="other_side" value="away" if="$(eval team_side == 'home')" />
    <arg name="other_side" value="home" if="$(eval team_side == 'away')" />
    
    <!-- Hardware agnostic code -->
    <node name="ai" pkg="teamrocket" type="ai_node.py" output="screen">
        <!-- <remap from="me"   to="orienter/$(arg team_side)$(arg my_number)" /> -->
        <!-- <remap from="ally" to="orienter/$(arg team_side)$(arg ally_number)" /> -->
        <!-- <remap from="opp1" to="orienter/$(arg other_side)1" /> -->
        <!-- <remap from="opp2" to="orienter/$(arg other_side)2" /> -->
        <!-- <remap from="ball" to="orienter/ball" /> -->
        <remap from="me"
               to="$(arg orienter_path)/$(arg team_side)$(arg my_number)" />
        <remap from="ally"
               to="$(arg orienter_path)/$(arg team_side)$(arg ally_number)" />
        <remap from="opp1" 
               to="$(arg orienter_path)/$(arg other_side)1" />
        <remap from="opp2" 
               to="$(arg orienter_path)/$(arg other_side)2" />
        <remap from="ball" 
               to="$(arg orienter_path)/ball" />
    </node>
    <node name="controller" pkg="teamrocket" type="controller_node.py" output="screen">
        <!-- The controller needs to know which orienter msg to listen to -->
        <remap from="me" to="/orienter/me$(arg my_number)" />
        <!-- For simulation, remap the output -->
        <remap from="vel_cmds" to="/$(arg team_side)$(arg my_number)/command" />
    </node>

</launch>
